manual/cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/MatchTest.scala

```diff
@@ -19,347 +19,305 @@
+        |       (michael)-[:ACTED_IN {role: "President Andrew Shepherd"}]->(thePresident),
+        |       (rob)-[:DIRECTED]->(thePresident)"""
     )
-  }
-
-  @Test def get_multiple_nodes() {
-    testQuery(
-      title = "Multiple starting points",
-      text = "By specifying two node patterns all combinations of the two nodes will be returned, so it is advised to add a constraint.",
-      queryText = """match (a), (b) where id(a) = id(b) return a, b""",
-      optionalResultExplanation = "Returns the combinations of the two nodes from the database.",
-      assertions = (p) => {
-        val allNodes: List[Node] = GlobalGraphOperations.at(db).getAllNodes.asScala.toList
-
-        assertEquals(allNodes, p.columnAs[Node]("a").toList)
+    synopsis("The `MATCH` clause is used to search for the pattern described in it.")
+    section("Introduction") {
+      p( """The `MATCH` clause allows you to specify the patterns Neo4j will search for in the database.
+           |This is the primary way of getting data into the current set of bindings.
+           |It is worth reading up more on the specification of the patterns themselves in <<introduction-pattern>>.""")
+      p( """`MATCH` is often coupled to a `WHERE` part which adds restrictions, or predicates, to the `MATCH` patterns, making them more specific.
+           |The predicates are part of the pattern description, not a filter applied after the matching is done.
+           |_This means that `WHERE` should always be put together with the `MATCH` clause it belongs to._""")
+      p( """`MATCH` can occur at the beginning of the query or later, possibly after a `WITH`.
+           |If it is the first clause, nothing will have been bound yet, and Neo4j will design a search to find the results matching the clause and any associated predicates specified in any `WHERE` part.
+           |This could involve a scan of the database, a search for nodes of a certain label, or a search of an index to find starting points for the pattern matching.
+           |Nodes and relationships found by this search are available as _bound pattern elements,_ and can be used for pattern matching of sub-graphs.
+           |They can also be used in any further `MATCH` clauses, where Neo4j will use the known elements, and from there find further unknown elements.""")
+      p( """Cypher is declarative, and so usually the query itself does not specify the algorithm to use to perform the search.
+           |Neo4j will automatically work out the best approach to finding start nodes and matching patterns.
+           |Predicates in `WHERE` parts can be evaluated before pattern matching, during pattern matching, or after finding matches.
+           |However, there are cases where you can influence the decisions taken by the query compiler.
+           |Read more about indexes in <<query-schema-index>>, and more about the specifying index hints to force Neo4j to use a specific index in <<query-using>>.""")
```

Comment:
I think the phrase 'index hints' is general enough to cover scan hints (but not join hints). Since we are not pushing join hints as a feature people should jump on, this text seems ok to me.